go/service/gregor.go

```diff
@@ -24,10 +24,14 @@ type gregorHandler struct {
 	cli              rpc.GenericClient
 	sessionID        gregor1.SessionID
 	skipRetryConnect bool
+	itemsByID        map[string]gregor.Item
```

Comment:
This can probably replaced by talking to the state machine after @mmaxim's work lands.